src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -1251,6 +1256,8 @@ def name(self):
 
     def matches_service(self, service_name: Optional[str]) -> bool:
         if service_name:
+            if self.daemon_type in ['haproxy', 'keepalived']:
+                return ("HA_RGW." + self.daemon_id).startswith(service_name + '.')
```

Comment:
this hack is a great reason we should avoid this `ha-rgw` == `haproxy` + `keepalived` split

Reply:
lets see  how can resolve this ... good discussion ahead :-)